src/server/game/Maps/Map.cpp

```diff
@@ -2596,6 +2600,104 @@ ZLiquidStatus Map::getLiquidStatus(float x, float y, float z, uint8 ReqLiquidTyp
     return result;
 }
 
+void Map::getFullTerrainStatusForPosition(float x, float y, float z, PositionFullTerrainStatus& data, uint8 reqLiquidType) const
+{
+    VMAP::IVMapManager* vmgr = VMAP::VMapFactory::createOrGetVMapManager();
+    VMAP::AreaAndLiquidData vmapData;
+    vmgr->getAreaAndLiquidData(GetId(), x, y, z, reqLiquidType, vmapData);
+    if ((data.hasFullAreaInfo = vmapData.hasAreaInfo))
+    {
+        data.areaInfo.adtId = vmapData.areaInfo.adtId;
+        data.areaInfo.rootId = vmapData.areaInfo.rootId;
+        data.areaInfo.groupId = vmapData.areaInfo.groupId;
+        data.areaInfo.mogpFlags = vmapData.areaInfo.flags;
+    }
+
+
+    GridMap* gmap = const_cast<Map*>(this)->GetGrid(x, y);
+    float mapHeight = gmap->getHeight(x, y);
+
+    // area lookup
+    AreaTableEntry const* areaEntry = nullptr;
+    if (vmapData.hasAreaInfo && !(z + 2.0f > mapHeight && mapHeight > vmapData.floorZ))
```

Comment:
dont bring out the negation before (), especially when you have conditions you can easily change (> to <=)